presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnValidator.java

```diff
@@ -49,33 +49,53 @@ public OrderableArrayColumnValidator()
         checkArgument(column.getType() instanceof ArrayType, "Expect ArrayType, found %s", column.getType().getDisplayName());
         Type elementType = ((ArrayType) column.getType()).getElementType();
 
-        FunctionCall arraySort = new FunctionCall(QualifiedName.of("array_sort"), ImmutableList.of(column.getIdentifier()));
         Expression checksum;
-        if (elementType instanceof ArrayType || elementType instanceof RowType) {
-            checksum = new CoalesceExpression(
-                    new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(new TryExpression(arraySort))),
-                    new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(column.getIdentifier())));
+        if (elementType.isOrderable()) {
+            FunctionCall arraySort = new FunctionCall(QualifiedName.of("array_sort"), ImmutableList.of(column.getIdentifier()));
+            if (elementType instanceof ArrayType || elementType instanceof RowType) {
+                checksum = new CoalesceExpression(
+                        new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(new TryExpression(arraySort))),
+                        new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(column.getIdentifier())));
+            }
+            else {
+                checksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(arraySort));
+            }
         }
         else {
-            checksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(arraySort));
+            checksum = new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(column.getIdentifier()));
         }
 
-        return ImmutableList.of(new SingleColumn(checksum, Optional.of(delimitedIdentifier(getChecksumColumnAlias(column)))));
+        FunctionCall arrayCardinality = new FunctionCall(QualifiedName.of("sum"),
+                ImmutableList.of(new FunctionCall(QualifiedName.of("cardinality"), ImmutableList.of(column.getIdentifier()))));
```

Comment:
First parameter on a separate line

Reply:
If all rows of the array column are null, `array(cardinality(a))` will be `null`, so let's wrap it with a `coalesce`.
```SQL
coalesce(array(cardinality(a)), 0) a_cardinality_sum
```

Reply:
Do you mean?
```sql
coalesce(sum(cardinality(a)), 0)
```

Reply:
Exactly. Sorry for the typo.